client/app/components/ParameterMappingInput.jsx

```diff
@@ -158,6 +159,13 @@ export class ParameterMappingInput extends React.Component {
       newMapping.param = newMapping.param.clone();
       newMapping.param.setValue(newMapping.value);
```

Comment:
Why doesn't this line already do the trick?
Cause in `StaticValue` `newMapping.value` is null?

Will #4049 fix this?

Reply:
That like wasn't doing the trick for this case because the `value` wasn't actually changing, but the `type` was. From the little I understood in the code, only type that should have a `value` is the Static. I think #4049 won't solve it because it has to do with the "mapping" instructions (and those are placed in the widget structure?).